apps/wallet-dashboard/lib/utils/migration.ts

```diff
@@ -49,3 +46,97 @@ export function groupStardustObjectsByMigrationStatus(
 
     return { migratable, unmigratable };
 }
+
+export function getTotalTimelockedAmount(stardustOutputObjects: IotaObjectData[]): number {
+    let totalTimelockedAmount = 0;
+
+    for (const outputObject of stardustOutputObjects) {
+        const outputObjectFields = extractOutputFields(outputObject);
+
+        totalTimelockedAmount += parseInt(outputObjectFields.balance);
+
+        totalTimelockedAmount += parseInt(outputObjectFields.balance);
+        const depositReturnAmount = calculateStorageDepositReturn(outputObjectFields);
+
+        if (depositReturnAmount) {
+            totalTimelockedAmount += depositReturnAmount;
+        }
+    }
+
+    return totalTimelockedAmount;
+}
+
+interface MigratableObjectsData {
+    totalNativeTokens: number;
+    totalVisualAssets: number;
+    accumulatedIotaAmount: number;
+}
+
+export function summarizeMigrationValues({
+    basicOutputObjects,
+    nftOutputObjects,
+    epochUnix,
+    address,
+}: {
+    basicOutputObjects: IotaObjectData[];
+    nftOutputObjects: IotaObjectData[];
+    epochUnix: number;
+    address: string;
+}): MigratableObjectsData {
+    let totalNativeTokens = 0;
+    let totalIotaAmount = 0;
+
+    const totalVisualAssets = nftOutputObjects.length;
+    const outputObjects = [...basicOutputObjects, ...nftOutputObjects];
+
+    for (const output of outputObjects) {
+        const outputObjectFields = extractOutputFields(output);
+
+        totalIotaAmount += parseInt(outputObjectFields.balance);
+        totalNativeTokens += parseInt(outputObjectFields.native_tokens.fields.size);
+
+        if (outputObjectFields.expiration_uc) {
+            const unlockableAddress = getUnlockableAddressFromUnlockConditions(
+                outputObjectFields.expiration_uc,
+                epochUnix,
+            );
+
+            // Only add the return amount if the unlockable address is the current address
+            if (unlockableAddress === address) {
+                const depositReturnAmount = calculateStorageDepositReturn(outputObjectFields) ?? 0;
+                totalIotaAmount += depositReturnAmount;
+            }
+        }
+    }
+
+    return { totalNativeTokens, totalVisualAssets, accumulatedIotaAmount: totalIotaAmount };
+}
+
+function calculateStorageDepositReturn({
+    storage_deposit_return_uc,
+}: CommonOutputObjectWithUc): number | undefined {
+    if (
+        storage_deposit_return_uc &&
+        'fields' in storage_deposit_return_uc &&
+        'return_amount' in storage_deposit_return_uc.fields
+    ) {
+        return parseInt(storage_deposit_return_uc?.fields.return_amount);
+    }
+}
+
+function extractOutputFields(outputObject: IotaObjectData): CommonOutputObjectWithUc {
+    return (
+        outputObject.content as unknown as {
+            fields: CommonOutputObjectWithUc;
+        }
+    ).fields;
+}
+
+function getUnlockableAddressFromUnlockConditions(
```

Comment:
each unlock condition has its return address
expiration unlock condition has a return address and SDRUC has another return address